tests/kademlia/testprovider.nim

```diff
@@ -0,0 +1,65 @@
+{.used.}
+from std/times import now, utc
+import chronicles
+import chronos
+import unittest2
+import ../../libp2p/[switch, builders]
+import ../../libp2p/protocols/kademlia/[kademlia, types]
+import ../utils/async_tests
+import ./utils.nim
+import ../helpers
+
+suite "KadDHT - AddProvider":
+  teardown:
+    checkTrackers()
+
+  asyncTest "Add provider":
+    var (switch1, kad1) = setupKadSwitch(PermissiveValidator(), CandSelector())
+    var (switch2, kad2) = setupKadSwitch(PermissiveValidator(), CandSelector())
+    var (switch3, kad3) = setupKadSwitch(PermissiveValidator(), CandSelector())
+    defer:
+      await allFutures(switch1.stop(), switch2.stop(), switch3.stop())
+
+    await kad1.bootstrap(@[switch2.peerInfo])
+    await kad3.bootstrap(@[switch2.peerInfo])
+
+    discard await kad1.findNode(kad2.rtable.selfId)
+    discard await kad3.findNode(kad2.rtable.selfId)
+
+    let
+      key = kad1.rtable.selfId
+      value = @[1.byte, 2, 3, 4, 5]
+
+    kad1.dataTable.insert(key, value, $times.now().utc)
+    kad2.dataTable.insert(key, value, $times.now().utc)
+    kad3.dataTable.insert(key, value, $times.now().utc)
+
+    check:
+      containsData(kad1, key, value)
+      containsData(kad2, key, value)
+      containsData(kad3, key, value)
```

Comment:
not sure if needed to be checked here. `insert` should be tested elsewhere, do don't need to assert same things all over the place. assume correctness if tests covers it.